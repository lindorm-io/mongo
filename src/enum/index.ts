export * from "./mongo-connection-type";
