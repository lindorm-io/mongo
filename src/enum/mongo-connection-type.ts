export enum MongoConnectionType {
  MEMORY = "MEMORY",
  STORAGE = "STORAGE",
}
