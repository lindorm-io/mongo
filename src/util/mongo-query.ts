import { IMongoConnectionOptions } from "../typing";
import { IRepositoryOptions } from "../repository";
import { Logger } from "@lindorm-io/winston";
import { MongoConnection } from "../class";

export type IDbQueryCallback = (options: IRepositoryOptions) => Promise<void>;

export interface IMongoQuery {
  mongoOptions: IMongoConnectionOptions;
  logger: Logger;
}

export const mongoQuery = async (options: IMongoQuery, callback: IDbQueryCallback): Promise<void> => {
  const mongo = new MongoConnection(options.mongoOptions);

  await mongo.connect();
  const db = await mongo.db();

  await callback({ db, logger: options.logger });

  await mongo.disconnect();
};

export const mongoPing = async (options: IMongoQuery): Promise<boolean> => {
  const mongo = new MongoConnection(options.mongoOptions);

  try {
    await mongo.connect();
    await mongo.disconnect();
    return true;
  } catch (_) {
    return false;
  }
};
