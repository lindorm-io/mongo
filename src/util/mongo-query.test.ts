import { mongoPing, mongoQuery } from "./mongo-query";
import { MongoConnectionType } from "../enum";
import { TObject } from "@lindorm-io/core";

describe("m", () => {
  let inMemoryStore: TObject<any>;
  let callback: any;
  let getMockOptions: any;

  beforeEach(() => {
    inMemoryStore = { initialized: true };

    callback = jest.fn();

    getMockOptions = () => ({
      mongoOptions: {
        type: MongoConnectionType.MEMORY,
        auth: {
          user: "user",
          password: "password",
        },
        url: {
          host: "host",
          port: 1234,
        },
        databaseName: "databaseName",
        inMemoryStore,
      },
      logger: "mock-logger",
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.resetAllMocks();
  });

  describe("mongoQuery", () => {
    test("should connect to mongo and use callback", async () => {
      await expect(mongoQuery(getMockOptions(), callback)).resolves.toBe(undefined);

      expect(callback).toHaveBeenCalledWith({
        db: {
          database: {},
          databaseName: "databaseName",
        },
        logger: "mock-logger",
      });
      expect(inMemoryStore).toMatchSnapshot();
    });
  });

  describe("mongoPing", () => {
    test("should return true if a connection can be established", async () => {
      await expect(mongoPing(getMockOptions())).resolves.toBe(true);
    });
  });
});
