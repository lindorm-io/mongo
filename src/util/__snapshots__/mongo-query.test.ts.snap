// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`m mongoQuery should connect to mongo and use callback 1`] = `
Object {
  "databaseName": Object {},
  "initialized": true,
}
`;
